test/integration/node-specific/bson-options/utf8_validation.test.ts

```diff
@@ -153,3 +156,177 @@ describe('class MongoDBResponse', () => {
     }
   );
 });
+
+describe('utf8 validation with cursors', function () {
+  let client: MongoClient;
+  let collection: Collection;
+
+  /**
+   * Inserts a document with malformed utf8 bytes.  This method spies on socket.write, and then waits
+   * for an OP_MSG payload corresponding to `collection.insertOne({ field: 'é' })`, and then modifies the
+   * bytes of the character 'é', to produce invalid utf8.
+   */
+  async function insertDocumentWithInvalidUTF8() {
+    const stub = sinon.stub(net.Socket.prototype, 'write').callsFake(function (...args) {
+      const providedBuffer = args[0].toString('hex');
+      const targetBytes = Buffer.from('é').toString('hex');
+
+      if (providedBuffer.includes(targetBytes)) {
+        if (providedBuffer.split(targetBytes).length !== 2) {
+          throw new Error('received buffer more than one `c3a9` sequences.  or perhaps none?');
```

Comment:
```suggestion
          sinon.restore();
          throw new Error('received buffer more than one `c3a9` sequences. or perhaps none?');
```

Maybe this is why we didn't see the error?

Reply:
Once this method returns / errors, we'll still have to continue up the sinon callstack back into the driver, regardless of whether or not the stub has been restored.